backend/common/services/s3_service.py

```
@@ -224,24 +232,38 @@ def parse_data_url(data_url: str) -> tuple[bytes, str, str]:
     image_bytes = base64.b64decode(base64_data)
     return image_bytes, content_type, ext
 
 
+def parse_generic_data_url(data_url: str) -> tuple[bytes, str]:
+    """Parse any data URL (not just images) and return (bytes, content_type)."""
+    import base64
+    import re
+
+    match = re.match(r'data:([^;]+);base64,(.+)', data_url, re.DOTALL)
+    if not match:
+        raise ValueError("Invalid data URL format")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)